### ISSUE TITLE
[SR-16057] Include target name in Linux test output.

### ISSUE THREAD

**Issue by adam-fowler** [apple/swift-corelibs-xctest#343]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-16057      |
   |Radar             | None         |
   |Original Reporter | @adam-fowler      |
   |Type              | Improvement    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | XCTest    |
   |Labels            | Improvement        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 796a6b7c95c2f0761a5fa1880bf3d0bf

  </details>

**Issue Description:**

When you run \`swift test\` on macOS it includes the target name in the output of each test.

 

``` java
Test Case '-[BasicsTests.AuthorizationProviderTests testBasicAPIs]' started.
Test Case '-[BasicsTests.AuthorizationProviderTests testBasicAPIs]' passed (0.001 seconds).

```

When you run the same on Linux it doesn't include the target name

 

 

``` java
Test Case 'AuthorizationProviderTests.testBasicAPIs' started at 2022-03-24 17:20:00.
Test Case 'AuthorizationProviderTests.testBasicAPIs' passed (0.001 seconds).
```

If you are parsing the output from the Linux version this makes it harder to work out what actual test passed/failed especially if your Package.swift has two targets which both have a test suite and function name with the same name.

Would it be possible to update the output to include the target name.

 

The context for this is I am parsing the output of \`swift test\` to update the status of Tests in the Test Explorer in the Swift Visual Studio Code Extension.

**Comment by tomerd** [comment]
@neonichu can the new output handler help here?

**Comment by neonichu** [comment]
I don't believe so, we can just pass along what we get as names for test cases from XCTest callbacks. I am assuming those values will be identical to what is being printed to the console.

**Comment by grynspan** [comment]
May need a compiler change to propagate the necessary information from the compiler build phase into the target.
